e3sm_diags/plot/utils.py

```diff
@@ -0,0 +1,459 @@
+import xcdat as xc
+from cartopy.mpl.ticker import LatitudeFormatter, LongitudeFormatter
+from matplotlib.transforms import Bbox
+
+from e3sm_diags.derivations.default_regions_xr import REGION_SPECS
+from e3sm_diags.driver.utils.general import get_output_dir
+from e3sm_diags.logger import custom_logger
+from e3sm_diags.parameter.core_parameter import CoreParameter
+from e3sm_diags.plot import get_colormap
+
+matplotlib.use("Agg")
+from matplotlib import colors  # isort:skip  # noqa: E402
+import matplotlib.pyplot as plt  # isort:skip  # noqa: E402
+
+logger = custom_logger(__name__)
+
+# Plot title and side title configurations.
+PLOT_TITLE = {"fontsize": 11.5}
+PLOT_SIDE_TITLE = {"fontsize": 9.5}
+
+# Position and sizes of subplot axes in page coordinates (0 to 1)
+PANEL = [
+    (0.1691, 0.6810, 0.6465, 0.2258),
+    (0.1691, 0.3961, 0.6465, 0.2258),
+    (0.1691, 0.1112, 0.6465, 0.2258),
+]
+
+# Border padding relative to subplot axes for saving individual panels
+# (left, bottom, right, top) in page coordinates
+BORDER_PADDING = (-0.06, -0.03, 0.13, 0.03)
+
+
+def _save_plot(fig: plt.figure, parameter: CoreParameter):
+    """Save the plot using the figure object and parameter configs.
+
+    This function creates the output filename to save the plot. It also
+    saves each individual subplot if the reference name is an empty string ("").
+
+    Parameters
+    ----------
+    fig : plt.figure
+        The plot figure.
+    parameter : CoreParameter
+        The CoreParameter with file configurations.
+    """
+    for f in parameter.output_format:
+        f = f.lower().split(".")[-1]
+        fnm = os.path.join(
+            get_output_dir(parameter.current_set, parameter),
+            parameter.output_file + "." + f,
+        )
+        plt.savefig(fnm)
+        logger.info(f"Plot saved in: {fnm}")
+
+    # Save individual subplots
+    if parameter.ref_name == "":
+        panels = [PANEL[0]]
+    else:
+        panels = PANEL
+
+    for f in parameter.output_format_subplot:
+        fnm = os.path.join(
+            get_output_dir(parameter.current_set, parameter),
+            parameter.output_file,
+        )
+        page = fig.get_size_inches()
+
+        for idx, panel in enumerate(panels):
+            # Extent of subplot
+            subpage = np.array(panel).reshape(2, 2)
+            subpage[1, :] = subpage[0, :] + subpage[1, :]
+            subpage = subpage + np.array(BORDER_PADDING).reshape(2, 2)
+            subpage = list(((subpage) * page).flatten())
+            extent = Bbox.from_extents(*subpage)
+
+            # Save subplot
+            fname = fnm + ".%i." % idx + f
+            plt.savefig(fname, bbox_inches=extent)
+
+            orig_fnm = os.path.join(
+                get_output_dir(parameter.current_set, parameter),
+                parameter.output_file,
+            )
+            fname = orig_fnm + ".%i." % idx + f
+            logger.info(f"Sub-plot saved in: {fname}")
+
+
+def _add_colormap(
```

Comment:
consider to change name to something like `_add_colormap_lat_lon`. Though it can be changed when other plotters being added, e.g. `zonal mean 2d` ...

Reply:
`_add_color_map()` is the refactored version of the old `lat_lon_plot.plot_panel()`, which `aerosol_aeronet_plot.plot()` also uses.

I can update the `_add_color_map` docstring to say it is used by `lat_lon_plot.py` and will be used by `aersol_aeronet_plot.py` to make it clearer.

More context:
https://github.com/E3SM-Project/e3sm_diags/blob/c02d7015bbd1b211bdf3165d8412ace29ca23d5c/e3sm_diags/plot/cartopy/deprecated_lat_lon_plot.py#L1-L7
https://github.com/E3SM-Project/e3sm_diags/blob/c02d7015bbd1b211bdf3165d8412ace29ca23d5c/e3sm_diags/plot/cartopy/deprecated_lat_lon_plot.py#L72-L74
https://github.com/E3SM-Project/e3sm_diags/blob/c02d7015bbd1b211bdf3165d8412ace29ca23d5c/e3sm_diags/plot/cartopy/aerosol_aeronet_plot.py#L10